controller/client/client.go

```diff
@@ -421,3 +423,29 @@ func (c *Client) ProviderList() ([]*ct.Provider, error) {
 	var providers []*ct.Provider
 	return providers, c.Get("/providers", &providers)
 }
+
+func (c *Client) Put(path string, in, out interface{}) error {
+	return c.send("PUT", path, in, out)
+}
+
+func (c *Client) Post(path string, in, out interface{}) error {
+	return c.send("POST", path, in, out)
+}
+
+func (c *Client) Get(path string, out interface{}) error {
+	return c.send("GET", path, nil, out)
+}
+
+func (c *Client) Delete(path string) error {
+	return c.send("DELETE", path, nil, nil)
+}
+
+func (c *Client) send(method, path string, in, out interface{}) (err error) {
+	for startTime := time.Now(); time.Since(startTime) < 10*time.Second; time.Sleep(100 * time.Millisecond) {
+		err = c.Send(method, path, in, out)
+		if !httphelper.IsPostgresError(err) {
```

Comment:
I'd prefer if the error was marked as a retry-able error instead of exposing internal details about the use of postgres.